src/rules/noUnusedVariableRule.ts

```diff
@@ -337,12 +339,14 @@ function forEachImport<T>(sourceFile: ts.SourceFile, f: (i: ImportLike) => T | u
     });
 }
 
-function findImport(pos: number, sourceFile: ts.SourceFile): ts.Identifier | undefined {
+function findImports(pos: number, sourceFile: ts.SourceFile, kind: UnusedKind): ReadonlyArray<ts.Identifier> {
```

Comment:
changing to an array since in the case of kind === DECLARATION we want to return all the imports in the node